### PR TITLE
New goal popup

### LINKED ISSUE #825: New goal popup
To set a goal user have to pick:
- how much time he/she wants to spend on it (30min, 45min, 1h, 1.5h, 2h, 3, 4, 5, 6, 7, 8, 9, 10h)
- on what he/she wants to spend time (pick a category)

Other notes:
- user can slide durations and categories from left to right and from right to left (similar how current editing categories works)
- tapping 'Set a goal' button takes users to Goals view

goal popup 1x

### PR DESCRIPTION
**Pull request by sofokli1**
Closes #825

### CODE REVIEW

File: teferi/UI/Common/TimeIntervalHelper.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import UIKit
```
Comment (Odrakir) [inline teferi/UI/Common/TimeIntervalHelper.swift]: This file shouldn't be part of this PR.

File: teferi/UI/Modules/Goals/GoalList/GoalViewModel.swift [hunk 2]
```diff
@@ -67,16 +67,21 @@ class GoalViewModel
     //MARK: Private Methods
     private func withMissingDateGoals(_ goals: [Goal]) -> [Goal]
     {
+        guard goals.count > 0 else { return [] }
```
Comment (Odrakir) [inline teferi/UI/Modules/Goals/GoalList/GoalViewModel.swift]: This method is getting too complicated for someone as dumb as me to understand. Wouldn't it be easier to just get the earliest goal and loop through all dates until present checking if that date is an array of goals? Just a suggestion...

File: teferi/UI/Modules/Goals/NewGoal/Views/HorizontalPicker.swift [hunk 3]
```diff
@@ -0,0 +1,300 @@
+import UIKit
+import RxSwift
+
+class HorizontalPicker<ItemType: Equatable> : UIControl, TrigonometryHelper, UIDynamicAnimatorDelegate
```
Comment (Odrakir) [inline teferi/UI/Modules/Goals/NewGoal/Views/HorizontalPicker.swift]: I know this probably comes from the time slot edit view (it was done before I joined Toggl). But... why do your own implementation instead of just using a `UICollectionView`? I think a solution with much less code and even with more beautiful result could be done by using collection views instead.

Comment (sofokli1) [reply teferi/UI/Modules/Goals/NewGoal/Views/HorizontalPicker.swift]: couldn't estimate how long it would have taken to do it with a collection view. reusing the same logic from other component seemed faster.
